### PR TITLE
[Ops][BugFix][MRV2] Fix apply bit mask kernel

### PR DESCRIPTION
**Pull request by zouzy5137**
### What this PR does / why we need it?
This PR aligns upstream function interface and fixes the `apply_grammar_bitmask` Triton kernel to correctly handle adaptive verification and per-request logit offsets. It introduces `cu_num_logits` to resolve absolute logit rows on device and handles inactive mapping entries safely.
### Does this PR introduce _any_ user-facing change?
No.
### How was this patch tested?
Tested with CI case.

- vLLM main: https://github.com/vllm-project/vllm/commit/e6bfe03ad73a3330cb427885aa90d97a12e1c704

### CODE REVIEW

**Review by AuroraEmiya — APPROVED** [review]

**Comment by zouzy5137** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E
- PR Create

**Comment by zouzy5137** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Review by Ronald1995 — APPROVED** [review]
LGTM

**Review by Tflowers-0129 — APPROVED** [review]

**Comment by Tflowers-0129** [comment]
LGTM.
